packages/commonwealth/server/controllers/server_communities_methods/create_community.ts

```diff
@@ -124,13 +124,15 @@ export async function __createCommunity(
     throw new AppError(Errors.NoBase);
   }
 
-  if (
-    community.icon_url &&
-    (await getFileSizeBytes(community.icon_url)) / 1024 > MAX_IMAGE_SIZE_KB
-  ) {
-    throw new AppError(Errors.ImageTooLarge);
+  if (community.icon_url) {
+    const iconFileSize = await getFileSizeBytes(community.icon_url);
+    if (iconFileSize === 0) {
+      throw new AppError(Errors.ImageDoesntExist);
+    }
+    if (iconFileSize / 1024 > MAX_IMAGE_SIZE_KB) {
+      throw new AppError(Errors.ImageTooLarge);
+    }
```

Comment:
should we encapsulate this and the code in createChain below inside a function receiving the icon url and returning the size or error?